client/internal/dns/host.go

```diff
@@ -19,6 +20,7 @@ type hostDNSConfig struct {
 }
 
 type domainConfig struct {
+	disabled  bool
```

Comment:
We are missing disabled domain config handling on Windows and Darwin

Reply:
Added, need help with testing.

Reply:
Tested in Windows - works as expected.

Reply:
Same for macOS